### PR TITLE
lib/ukboot: Fix configuration of no "no scheduler"

### PR DESCRIPTION
**Pull request by skuenzer**
### Description of changes

This PR fixes the KConfig configuration of "no scheduler" with `lib/ukboot`. Previously, scheduler initialization code was still compiled in even if no scheduling was selected. This happened if at the same time no allocator was configured as well.

Please note that this PR fixes only the configuration handling in `lib/ukboot`. I noticed a crash within one of the registered inittab functions if no scheduler was initialized. There is also an assertion failure if `lib/ukboot` is configured without allocator initialization (system does not boot without an allocator assigned to the platform libraries). Both need fixing with corresponding subsystem PRs and taking re-arch into account.

### CODE REVIEW

**Review by michpappas — APPROVED** [review]
Looks good. In the future we need to be a bit careful not to use `LIBUKBOOT_NOALLOC` /`LIBUKBOOT_NOSCHED`, also during reviews. I will wait for one more RB before approving for merge.

Reviewed-by: Michalis Pappas <michalis@unikraft.io>

**Review by StefanJum — APPROVED** [review]
Works fine, thank you.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**
